test/common/stats/lazyinit_stats_speed_test.cc

```diff
@@ -0,0 +1,168 @@
+#include "envoy/upstream/upstream.h"
+
+#include "source/common/common/random_generator.h"
+#include "source/common/stats/isolated_store_impl.h"
+#include "source/common/stats/lazy_init.h"
+#include "source/common/stats/symbol_table.h"
+#include "source/common/stats/thread_local_store.h"
+#include "source/common/thread_local/thread_local_impl.h"
+#include "source/exe/process_wide.h"
+
+#include "test/benchmark/main.h"
+#include "test/common/stats/real_thread_test_base.h"
+#include "test/test_common/real_threads_test_helper.h"
+
+#include "benchmark/benchmark.h"
+
+namespace Envoy {
+
+namespace Stats {
+
+using Upstream::ClusterTrafficStats;
+
+class LazyInitStatsBenchmarkBase {
+public:
+  LazyInitStatsBenchmarkBase(bool lazy, const uint64_t n_clusters, Stats::Store& s)
+      : lazy_init_(lazy), num_clusters_(n_clusters), stat_store_(s),
+        stat_names_(stat_store_.symbolTable()) {}
+
+  void createStats(bool defer_init) {
+    for (uint64_t i = 0; i < num_clusters_; ++i) {
+      std::string new_cluster_name = absl::StrCat("cluster_", i);
+      auto scope = stat_store_.createScope(new_cluster_name);
```

Comment:
here I'd use ScopeSharedPtr rather than `auto` as it's not obvious from call-site.